######################### Grasp Form Search

link_length: 0.6 # hydrus3: 0.44 
duct_radius: 0.2 # hydrus3: 0.15 #real is + 0.9(sponge)
joint_states_pub_name: /hydrusx/joint_states

# transform controller
kinematic_verbose: false
control_verbose: false
f_max: 15
f_min: 2

######################### Grasp Planning

joint_motors_sub_name: /motor_states/joints_port
joint_ctrl_pub_name: /hydrusx/joints_ctrl
joint_states_sub_name: /hydrusx/joint_states

aerial_transportation_plugin_name: aerial_transportation/hydrus

nav_vel_limit: 0.3
vel_nav_threshold: 0.2

object_approach_offset_x: -0.15
object_approach_offset_y: 0.3

approach_pos_threshold: 0.1
approach_count: 1.0
object_head_direction: true

object_height: 0.035 # set to the grasp height
grasping_height_threshold: 0.005
grasping_height_offset: 0.0

ascending_speed: 0.1
transportation_threshold: 0.15
transportation_count: 1.0

torque_min_threshold: 0.3
torque_max_threshold: 0.8 #deprecated

modification_delta_angle: 0.01

# grasp config
tighten_delta_angle: 0.10 #hydrusx_quad: box, cylinder: 0.1; hydrus3: box, cylinder: 0.14
grasping_rate: 4.0 #hydrusx_quad: box cylinder: 4.0; hydrus3: box: 2.0, cylinder: 1.0?

# approach config
approach_base_link: 2 #approaching start from 0; link2(==1) for hydrus3; link3(==2) for hydrusx_quad
approach_delta_angle: -0.4
approach_pos_weight_rate: 10 #hydrusx: box: 10 cylinder: 10 (4 maybe better?)  old: 10.0 #weight algorithm
approach_angle_weight_rate: 100.0 #weight algorithm

# no velocity control gain change if uncomment following line
#aerial_grasping_flight_velocity_control_pub_name: none
